### PR TITLE
Fix Synergy building on macOS

### PR DESCRIPTION
**Pull request by Enzime**
Update gtest and gmock to fix clang compiler issues, as well as patch CMakeLists.txt in multiple places to fix other issues on macOS.

###### Things done

- [x] Tested without sandboxing on macOS
- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Enzime** [comment]
@GrahamcOfBorg build synergy

**Review by adisbladis — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/synergy/default.nix [hunk 1]
```diff
@@ -14,17 +12,30 @@ stdenv.mkDerivation rec {
     sha256 = "0ksgr9hkf09h54572p7k7b9zkfhcdb2g2d5x7ixxn028y8i3jyp3";
   };
 
-  patches = [ ./openssl-1.1.patch ];
+  patches = [./openssl-1.1.patch ./update_gtest_gmock.patch
```
[the remark this answers is not in the document]
Comment (Enzime) [reply pkgs/applications/misc/synergy/default.nix]: If this is a patch I made for Nixpkgs, where should I host it?

[the remark this answers is not in the document]
Comment (adisbladis) [reply pkgs/applications/misc/synergy/default.nix]: My bad. I didn't notice these patches were made by you.
I assumed they were from upstream.

File: pkgs/applications/misc/synergy/default.nix [hunk 2]
```diff
@@ -14,17 +12,30 @@ stdenv.mkDerivation rec {
...
+  ] ++ lib.optional stdenv.isDarwin ./respect_macos_arch.patch;
 
   patch_gcc6 = fetchpatch {
     url = https://raw.githubusercontent.com/gentoo/gentoo/20e2bff3697ebf5f291e9907b34aae3074a36b53/dev-cpp/gmock/files/gmock-1.7.0-gcc6.patch;
     sha256 = "0j3f381x1lf8qci9pfv6mliggl8qs2w05v5lw3rs3gn7aibg174d";
   };
 
+  gmock_zip = fetchurl {
```
[the remark this answers is not in the document]
Comment (Enzime) [reply pkgs/applications/misc/synergy/default.nix]: Added a comment noting that it was only related to this version of Synergy.

File: pkgs/applications/misc/synergy/default.nix [hunk 3]
```diff
@@ -38,15 +49,19 @@ stdenv.mkDerivation rec {
...
     #
     # So let's force it:
-  + optionalString stdenv.isLinux ''
+  + lib.optionalString stdenv.isLinux ''
     sed -i -e '/HAVE_X11_EXTENSIONS_XRANDR_H/c \
       set(HAVE_X11_EXTENSIONS_XRANDR_H true)
     ' CMakeLists.txt
   '';
 
+  cmakeFlags = [ "-DOSX_TARGET_MAJOR=10" "-DOSX_TARGET_MINOR=7" ];
```
[the remark this answers is not in the document]
Comment (Enzime) [reply pkgs/applications/misc/synergy/default.nix]: Fixed.

File: pkgs/applications/misc/synergy/default.nix [hunk 4]
```diff
@@ -38,15 +52,19 @@ stdenv.mkDerivation rec {
...
     #
     # So let's force it:
-  + optionalString stdenv.isLinux ''
+  + lib.optionalString stdenv.isLinux ''
     sed -i -e '/HAVE_X11_EXTENSIONS_XRANDR_H/c \
       set(HAVE_X11_EXTENSIONS_XRANDR_H true)
     ' CMakeLists.txt
   '';
 
+  cmakeFlags = lib.optionals stdenv.isDarwin [ "-DOSX_TARGET_MAJOR=10" "-DOSX_TARGET_MINOR=7" ];
```
[the remark this answers is not in the document]
Comment (Enzime) [reply pkgs/applications/misc/synergy/default.nix]: Synergy's CMakeLists.txt expects these variables, and doesn't infer from `MACOSX_DEPLOYMENT_TARGET`.

File: pkgs/applications/misc/synergy/default.nix [hunk 5]
```diff
@@ -38,15 +52,19 @@ stdenv.mkDerivation rec {
...
+
   buildInputs = [
-    cmake xlibsWrapper libX11 libXi libXtst libXrandr xinput curl openssl
-  ];
+    cmake curl openssl
+  ] ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks;
```
[the remark this answers is not in the document]
Comment (Enzime) [reply pkgs/applications/misc/synergy/default.nix]: Fixed.

**Comment by grahamc** [comment]
@GrahamcOfBorg build synergy

**Review by LnL7 — APPROVED** [review]

**Pull request merged**